Standards/scs-XXXX-vN-security-groups-decision-record.md

```diff
@@ -0,0 +1,130 @@
+That creates some diffucilties for a possible standard of Security Groups, which are discussed in this document.
+
+## Terminology
+
+Security Group
+  A set of iptables rules that is applied to ports connecting a VM and a network.
+
+Security Group Rule (abbr. Rule)
+  This references a single rule within a security group.
+
+RBAC
+  Role Based Access Control used for policies and alike.
+
+network rbac / neutron rbac
+  These access controls will let administrators and users share neutron related resources with other projects.
+
+admin
+  The most powerful role in OpenStack. Only given to some Operators of the Cloud Infratructure.
+
+## Context
+
+While creating a VM and also later on, one or more security groups can be added to it.
+When there is no security group specified the default security group will always be added.
+Like every other security group, the default group is also project bound.
+That means, it can be edited as required by project members.
+
+When thinking about standardizing security groups, the goal is to automatically have more than one default security group.
+But to also have pre-defined groups with predefined roles for certain and often used cases as e.g. ssh, http and https (maybe also icmp).
+Those groups should best be available to all projects and pre-defined by administrators.
+
+As security groups are project bound and there is no native way to them to be shared, we are left with two options:
+
+1. To use another endpoint `network rbac` to share security groups among different projects.
+2. To adhere to the project scope of security groups and only give documentation about recommended security groups to users.
+
+### Option 1: operator usage of network rbac
+
+The `network rbac` endpoint[^1] manages the possibitity to share and access certain network sepcific resources as security groups.
+For admins it is possible to use this endpoint to share a security group with ALL projects within the the cloud including ALL projects of ALL domains:
+
+```bash
+stack@devstack:~/devstack$ openstack network rbac create --target-all-projects --action access_as_shared --type security_group group-for-everyone
++-------------------+--------------------------------------+
+| Field             | Value                                |
++-------------------+--------------------------------------+
+| action            | access_as_shared                     |
+| id                | 394d8e71-143f-4c5b-a72f-cd10af3222df |
+| object_id         | b6a3834a-f89c-47a9-9ed6-ca89e93701c4 |
+| object_type       | security_group                       |
+| project_id        | 15f2ab0eaa5b4372b759bde609e86224     |
+| target_project_id | *                                    |
++-------------------+--------------------------------------+
+```
+
+This would fulfill our goal to grant access to pre-defined security groups fro all projects and all groups recieved as shared do not count into the projects quota for security groups.
```

Comment:
```suggestion
This would fulfill our goal to grant access to predefined security groups fro all projects and all groups recieved as shared do not count into the projects quota for security groups.
```